### PR TITLE
Shorten Participants constructor

### LINKED ISSUE #88: Shorten Maven...Participant's constructors
The constructor for the "participants" (eg MavenCompletionParticipant) is pretty big. It should probably be modified to have less constructor paramaters. See https://github.com/eclipse/lemminx-maven/pull/78#discussion_r434111954

### PR DESCRIPTION
**Pull request by AObuchow**
Fix #88

Signed-off-by: Andrew Obuchowicz <aobuchow@redhat.com>

### CODE REVIEW

File: lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenCompletionParticipant.java [hunk 1]
```diff
@@ -105,12 +105,12 @@ public NodeWithChildrenInsertionStrategy(String elementName) {
 	private final MavenSession mavenSession;
 	private final BuildPluginManager buildPluginManager;
 
-	public MavenCompletionParticipant(MavenProjectCache cache, LocalRepositorySearcher localRepositorySearcher, RemoteRepositoryIndexSearcher indexSearcher, RepositorySystemSession repoSession, MavenSession mavenSession, MavenPluginManager pluginManager, BuildPluginManager buildPluginManager) {
+	public MavenCompletionParticipant(MavenProjectCache cache, LocalRepositorySearcher localRepositorySearcher, RemoteRepositoryIndexSearcher indexSearcher, MavenSession mavenSession, MavenPluginManager pluginManager, BuildPluginManager buildPluginManager) {
```
Comment (AObuchow) [inline lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenCompletionParticipant.java:108]: @mickaelistria this didn't shorten the constructors as much as I had hoped. What do you think about making a container class to hold things such as the MavenProjectCache, LocalRepositorySearcher,  RemoteRepositoryIndexSearcher, MavenSession etc.

Comment (mickaelistria) [reply lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenCompletionParticipant.java:108]: IMO, it's not worth the effort. A long constructor list isn't a too bad thing per se as it doesn't cause bugs, doesn't have a performance cost...; so introducing more code and more classes without clear design gain is not really helping.

Comment (mickaelistria) [reply lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenCompletionParticipant.java:108]: That said, the current change is very good as it removes some useless args we can easily deduce from other ones.

Comment (AObuchow) [reply lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenCompletionParticipant.java:108]: Okay sounds good with me. Will merge.

**Pull request merged**